src/Commands/CommandCollector.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Minicli\Framework\Commands;
+
+final class CommandCollector
+{
+    public function __construct(
+        public array $commands = [],
+    ) {}
+
+    public function add(
+        string $signature,
+        string $description,
+        string $instance,
+    ): void {
+        $this->commands[$signature] = [
```

Comment:
I think it's better to create a class to wrap these properties, so $commands can be an array of CommandWrapper or something like that.

Reply:
Yeah this is something I wasn't sure on - I like the idea though! Something like:

```php
final class StoredCommand
{
    public function __construct(
        private readonly string $signature,
        ....
    ) {}
}
```

Reply:
I like it, if you think that could be a nice thing, go for it!!!